### ISSUE TITLE
medusa-js client - GET requests includes a body `{}` which doesn't work well with AWS Cloudfront

### ISSUE THREAD

**Issue by nielswart** [medusajs/medusa#1043]
## Issue
GET requests includes a body, which is not allowed when putting the API behind a CDN such as AWS Cloudfront.  This results in a 403 error.

Looking at the request in the Medusa Client class, I see that the default payload is `{}`

## Possible Solution

make the default payload null or undefined

```
async request(
    method: RequestMethod,
    path: string,
    payload: object = null,
    options: RequestOptions = {},
    customHeaders: object = {}
  ): Promise<any> 
```

**Comment by srindom** [comment]
Thanks for reporting this - makes sense :) 

I think this is the culprit: https://github.com/medusajs/medusa/blob/master/packages/medusa-js/src/request.ts#L186

Would you be interested in PRing?

**Comment by nielswart** [comment]
> I think this is the culprit: https://github.com/medusajs/medusa/blob/master/packages/medusa-js/src/request.ts#L186
> 
> Would you be interested in PRing?

Thanks @srindom 

Yeah, thought so too.  I'll create a PR, thanks.  Do you agree that setting the payload object to `null` is the appropriate fix.  I'm not so familiar with the codebase yet.

**Comment by srindom** [comment]
Yeah I think that should be okay - as long as it doesn't affect the `POST` endpoints that aren't expecting a payload I see no issues. You can maybe test `POST /draft-orders/:id/pay` :)

**Comment by olivermrbl** [comment]
Solved in #1044
